### ISSUE TITLE
Termux on Adnroid multiprocessing not supported

### ISSUE THREAD

**Issue by DaTiC0** [anishathalye/git-remote-dropbox#88]
Can you somehow disable multiprocessing on Termux? because Android phones does not support it?
**ERROR CODE:**

```
Traceback (most recent call last):
  File "/data/data/com.termux/files/usr/lib/python3.7/multiprocessing/synchroniz                                          e.py", line 28, in <module>
    from _multiprocessing import SemLock, sem_unlink
ImportError: cannot import name 'SemLock' from '_multiprocessing' (/data/data/co                                          m.termux/files/usr/lib/python3.7/lib-dynload/_multiprocessing.cpython-37m.so)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/data/com.termux/files/usr/bin/git-remote-dropbox", line 10, in <mo                                          dule>
    sys.exit(main())
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/git_remote_d                                          ropbox/__init__.py", line 812, in main
    helper.run()
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/git_remote_d                                          ropbox/__init__.py", line 375, in run
    self._do_fetch(line)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/git_remote_d                                          ropbox/__init__.py", line 422, in _do_fetch
    self._fetch(sha)
  File "/data/data/com.termux/files/usr/lib/python3.7/site-packages/git_remote_d                                          ropbox/__init__.py", line 610, in _fetch
    input_queue = multiprocessing.Queue()  # requesting downloads
  File "/data/data/com.termux/files/usr/lib/python3.7/multiprocessing/context.py                                          ", line 102, in Queue
    return Queue(maxsize, ctx=self.get_context())
  File "/data/data/com.termux/files/usr/lib/python3.7/multiprocessing/queues.py"                                          , line 39, in __init__
    from .synchronize import SEM_VALUE_MAX as maxsize
  File "/data/data/com.termux/files/usr/lib/python3.7/multiprocessing/synchroniz                                          e.py", line 32, in <module>
    " synchronization primitives needed will not" +
ImportError: This platform lacks a functioning sem_open implementation, therefore, the required synchronization primitives needed will not function, see issue 3770.
```

**Comment by anishathalye** [comment]
Oh interesting, I didn't know anyone was using this on Android phones. And I didn't know that the `multiprocessing` module isn't fully supported on there.

This should be possible to fix by falling back to sequential execution for uploading/downloading files, though it might be a bit slow. I don't know if there's a better intermediate solution (e.g. something like `multiprocessing.Queue` that can be used on your platform).

**Comment by harounb** [comment]
Using https://github.com/asciinema/asciinema/issues/271#issuecomment-379389020 as a reference I've fixed the cloning portion of it in my fork:
https://github.com/harounb/git-remote-dropbox/tree/fix-termux.
Yet to test the other parts that rely on multiprocessing modules like pushing. I'll do that soon.

**Comment by anishathalye** [comment]
Fixed in #89.
